libra-node/src/main_node.rs

```diff
@@ -162,12 +164,33 @@ pub fn setup_environment(node_config: &mut NodeConfig) -> LibraHandle {
 
         reconfig_subscriptions.append(network_builder.reconfig_subscriptions());
 
-        // Start the network and cache the runtime so it does not go out of scope.
-        // TODO:  move all 'start' commands to a second phase at the end of setup_environment.  Target is to have one pass to wire the pieces together and a second pass to start processing in an appropriate order.
-        let peer_id = network_builder.peer_id();
-        let _listen_addr = network_builder.build();
+        network_builders.push(network_builder);
+    }
+
+    // Build the configured networks.
+    for network_builder in &mut network_builders {
+        debug!(
+            "Creating runtime for {:?}",
+            network_builder.network_context()
+        );
+        let runtime = Builder::new()
+            .thread_name("network-")
+            .threaded_scheduler()
+            .enable_all()
+            .build()
+            .expect("Failed to start runtime. Won't be able to start networking.");
+        network_builder.build(runtime.handle().clone());
         network_runtimes.push(runtime);
-        debug!("Network started for peer_id: {}", peer_id);
+        debug!(
+            "Network Built for network_context: {:?}",
+            network_builder.network_context()
+        );
+    }
+
+    // Start the configured networks.
+    // TODO:  Collect all component starts at the end of this function
+    for network_builder in &mut network_builders {
```

Comment:
any reason to not just merge these two? the networks don't seem to have any overlap.. and that way we can unify the log into built -> start, right now we have no long for starting... so maybe something weird happened between at starting

Reply:
Other components lack support for separating create/build/start.   It is technically meaningless to separate the pieces for network and not the others.

Stylistically, performing the (meaningless) separation now at least points to the direction and may encourage people to think in that way moving forward).